src/main/java/htsjdk/samtools/util/AbstractRecordAndOffset.java

```diff
@@ -91,24 +87,17 @@ public byte getReadBase() {
     }
 
     /**
-     * @return the base quality according to <code>offset</code>.
-     */
-    public byte getBaseQuality() {
-        return record.getBaseQualities()[offset];
-    }
-
-    /**
      * @return the length of alignment block represented by the object.
      */
     public int getLength() {
-        return length;
+        return 1;
     }
 
     /**
      * @return the position in reference sequence, to which the start of alignment block is aligned.
      */
     public int getRefPos() {
-        return refPos;
+        return 0;
```

Comment:
zero as an error code? better use -1 to avoid confusing with zero-based indexing.